From fa7f899f2b9faa9d1ae2752449ece6761e570557 Mon Sep 17 00:00:00 2001
From: Dan McDonald <danmcd@joyent.com>
Date: Mon, 28 Oct 2019 13:58:23 -0400
Subject: [PATCH] OS-8035 ipf -l pass causes panic in fr_dolog Reviewed by:
 Jason King <jason.king@joyent.com> Reviewed by: John Levon
 <john.levon@joyent.com> Reviewed by: Michael Zeller <mike.zeller@joyent.com>
 Approved by: Mike Gerdts <mike.gerdts@joyent.com>

---
 usr/src/uts/common/inet/ipf/ip_log.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/usr/src/uts/common/inet/ipf/ip_log.c b/usr/src/uts/common/inet/ipf/ip_log.c
index 7f5a1c839d..b70e320def 100644
--- a/usr/src/uts/common/inet/ipf/ip_log.c
+++ b/usr/src/uts/common/inet/ipf/ip_log.c
@@ -373,14 +373,15 @@ u_int flags;
 	if (fin->fin_fr != NULL) {
 		ipfl.fl_loglevel = fin->fin_fr->fr_loglevel;
 		ipfl.fl_logtag = fin->fin_fr->fr_logtag;
+		bcopy(fin->fin_fr->fr_uuid, ipfl.fl_uuid, sizeof (uuid_t));
 	} else {
 		ipfl.fl_loglevel = 0xffff;
 		ipfl.fl_logtag = FR_NOLOGTAG;
+		bzero(ipfl.fl_uuid, sizeof (uuid_t));
 	}
 	if (fin->fin_nattag != NULL)
 		bcopy(fin->fin_nattag, (void *)&ipfl.fl_nattag,
 		      sizeof(ipfl.fl_nattag));
-	bcopy(fin->fin_fr->fr_uuid, ipfl.fl_uuid, sizeof (ipfl.fl_uuid));
 	ipfl.fl_flags = flags;
 	ipfl.fl_dir = fin->fin_out;
 	ipfl.fl_lflags = fin->fin_flx;
-- 
2.17.2 (Apple Git-113)

